From abb18e1442575b6c255a4b6c9ea60180259df084 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 07 Sep 2022 20:26:33 +0300
Subject: [PATCH] [SW-0] gaudi3: cleanup boot ptr ctx structure

This patch removes "max_imgs", which is a redundant member of
boot ptr ctx structure. This is redundant because the rom code
is updated to support extension of number of images supported
for both rom and erom image pointers.

Change-Id: Ib3b01686467f86958b773d302df8c7196b26a3db
---

diff --git a/common/hl_boot_desc_rom.h b/common/hl_boot_desc_rom.h
index 6be9424..81af8fa 100644
--- a/common/hl_boot_desc_rom.h
+++ b/common/hl_boot_desc_rom.h
@@ -11,11 +11,6 @@
 #include <stdint.h>
 #include <hl_boot_desc.h>
 
-#define HL_MAX_IMG_PTR_IMGS	((uint8_t)(HL_FW_IMG_SPI_MAX_BOOT_PTRS) >\
-				(uint8_t)(HL_FW_MAX_EROM_BOOT_PTRS)\
-				? HL_FW_IMG_SPI_MAX_BOOT_PTRS :\
-				HL_FW_MAX_EROM_BOOT_PTRS)
-
 enum hl_fw_img_type_erom {
 	HL_FW_EROM_IMG_ROOT = 0,
 	HL_FW_EROM_PCIE = 1,
@@ -32,7 +27,6 @@
 
 struct hl_fw_boot_ptrs_ctx {
 	uint8_t boot_desc_flag;
-	uint8_t max_imgs;
 	uint32_t img_ptr_spi_offset;
 	uint64_t start_addr;
 	struct hl_fw_desc_info desc_info;
